client/src/components/Friends/FriendRequestsComponent.js

```diff
@@ -0,0 +1,280 @@
+import React, { useState, useEffect } from 'react'
+import { makeStyles } from '@material-ui/core/styles'
+import axios from 'axios'
+
+import {
+  Typography,
+  ListItem,
+  ListItemAvatar,
+  Avatar,
+  List,
+  ListItemText,
+  ListItemSecondaryAction,
+  IconButton,
+  Tooltip
+} from '@material-ui/core'
+
+import CheckCircleIcon from '@material-ui/icons/CheckCircle'
+import CancelIcon from '@material-ui/icons/Cancel'
+
+const useStyles = makeStyles(theme => ({
+  noResults: {
+    textAlign: 'center',
+    color: 'gray'
+  },
+  avatar: {
+    border: '1px solid black'
+  }
+}))
+
+const FriendRequestsComponent = ({
+  accountValues,
+  relations,
+  relationsUsers,
+  reloadRelations,
+  handleIncomingFriendRequestCountChange
+}) => {
+  const classes = useStyles()
+
+  const [incomingRequests, setIncomingRequests] = useState([])
+  const [sentRequests, setSentRequests] = useState([])
+
+  const handleAcceptRequest = relation => {
+    axios
+      .patch('/users/add', {
+        userId1: accountValues.id,
+        userId2: relation.id
+      })
+      .then(() => {
+        reloadRelations()
+      })
+      .catch(() => {})
+  }
+
+  const handleRemoveRequest = relation => {
+    const userId1 = accountValues.id
+    const userId2 = relation.id
+
+    // DELETE requests have to pass to data instead of regular payload
+    axios
+      .delete('/users/remove', {
+        data: {
+          userId1: userId1,
+          userId2: userId2
+        }
+      })
+      .then(() => {
+        reloadRelations()
+      })
+      .catch(() => {})
+  }
+
+  useEffect(() => {
+    if (relations && relationsUsers) {
+      // Set incoming requests array by combining two filtered portions of the relation array
+      // (Logged in user is being sent the friend request from someone else,
+      // which is Id1 for FRIEND_REQUEST_2_TO_1 or Id2 for FRIEND_REQUEST_1_TO_2)
+      // and concatenating the name and imageUrl information through cross-referencing to users array
+      setIncomingRequests([
+        ...relations
+          .filter(
+            relation =>
+              relation.userId1 === accountValues.id &&
+              relation.status === 'FRIEND_REQUEST_2_TO_1'
+          )
+          .map(filteredRelation => {
+            let relationUser = relationsUsers.find(
+              user => user._id === filteredRelation.userId2
+            )
+
+            if (!relationUser || Object.keys(relationUser).length === 0)
+              return {}
+
+            return {
+              id: filteredRelation.userId2,
+              message: filteredRelation.message,
+              name: relationUser.name,
+              imageUrl: relationUser.imageUrl
+            }
+          }),
```

Comment:
this should be okay, but generally when there's a .filter().map(), it can be replaced with a .reduce()

Reply:
I generally like to follow best practices, but this code relies on many variables that are dependent on async functions so I'm a bit scared to try and refactor this for the time being...